peer/x/roundrobin/list.go

```diff
@@ -55,27 +55,28 @@ type List struct {
 	started            atomic.Bool
 }
 
-func (pl *List) addAll(peerIDs []peer.Identifier) error {
+// Update applies the additions and removals of peer Identifiers to the list
+// it returns a multi-error result of every failure that happened without
+// circuit breaking due to failures
+func (pl *List) Update(additions, removals []peer.Identifier) error {
 	pl.lock.Lock()
 	defer pl.lock.Unlock()
 
 	var errs []error
 
-	for _, peerID := range peerIDs {
+	for _, peerID := range additions {
 		if err := pl.addPeerIdentifier(peerID); err != nil {
 			errs = append(errs, err)
 		}
 	}
 
-	return yerrors.MultiError(errs)
-}
+	for _, peerID := range removals {
+		if err := pl.removePeerIdentifier(peerID); err != nil {
+			errs = append(errs, err)
+		}
+	}
```

Comment:
Remove before add will be safer if a provider adds and removes the same peer. It will remain correct (but flap) instead of erring and then losing the peer outright.